extensions/query-engine/src/main/java/com/aerospike/helper/query/cache/IndexCache.java

```diff
@@ -0,0 +1,57 @@
+package com.aerospike.helper.query.cache;
+
+import com.aerospike.client.AerospikeClient;
+import com.aerospike.client.Info;
+import com.aerospike.client.cluster.Node;
+import com.aerospike.client.policy.InfoPolicy;
+import com.aerospike.helper.model.Index;
+import com.aerospike.helper.query.QueryEngine;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class IndexCache implements AutoCloseable {
+
+    private static final String SINDEX = "sindex";
+    private volatile Map<IndexKey, Index> indexCache = Collections.emptyMap();
```

Comment:
I replaced TreeMap with simple HashMap as I don't see there a need for sorted map. Did I miss something?

Reply:
we do not need ConcurrentHashMap here as we are swapping links on refresh and the contents of the map do not change in runtime. `IndexKey` is already immutable. I missed making `Index` class immutable and wrapping map into `Collections.unmodifiableMap`. these changes will guarantee immutability of the map -- so no concurrency needed for the contents of it.

Reply:
It should be alright from caller side, but this will open the possibility of multiple concurrent Info calls which server might have a hard time to handle.